### PR TITLE
feat(explore): Fill dashboard name when adding new chart from dashboard view

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
When user enters chart creation flow starting from a dashboard (e.g. with "Create chart" button in empty state or charts list in dashboard edit mode), the chart's save modal should have that dashboard's name prefilled. To achieve that, we pass `dashboard_id` as a url param to `/chart/add` page and from there to `explore`.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

https://user-images.githubusercontent.com/15073128/169276098-5a2578a0-7ed4-4029-aa40-2f112e3af13a.mov

### TESTING INSTRUCTIONS
1. Open a dashboard -> edit mode -> charts tab -> click "Create new chart" OR Open a dashboard and create a new chart using the plus icon button in top right corner.
2. Create a chart and click save
3. Verify that dashboard's name is prefilled
4. Create a new dashboard and click "Create new chart" button in empty state. Create a chart and verify that dashboard's name is prefilled in save modal
5. Create a chart from any other view (for example from Welcome page). Verify that save modal is not prefilled

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC @kasiazjc

### CODE REVIEW

File: superset-frontend/src/explore/components/ExploreViewContainer/index.jsx [hunk 1]
```diff
@@ -699,12 +699,17 @@ function mapStateToProps(state) {
   const chartKey = Object.keys(charts)[0];
   const chart = charts[chartKey];
 
+  let dashboardId = Number(explore.form_data?.dashboardId);
+  if (Number.isNaN(dashboardId)) {
```
Comment (kgabryje) [inline superset-frontend/src/explore/components/ExploreViewContainer/index.jsx:703]: `??` operator doesn't exclude NaNs. I also don't want to use `||` operator in order not to exclude dashboards with id `0`

File: superset-frontend/src/addSlice/AddSliceContainer.tsx [hunk 2]
```diff
@@ -195,10 +198,12 @@ export default class AddSliceContainer extends React.PureComponent<
   }
 
   exploreUrl() {
+    const dashboardId = getUrlParam(URL_PARAMS.dashboardId);
     const formData = encodeURIComponent(
       JSON.stringify({
         viz_type: this.state.visType,
         datasource: this.state.datasource?.value,
+        ...(!isNullish(dashboardId) && { dashboardId }),
```
Comment (AAfghahi) [inline superset-frontend/src/addSlice/AddSliceContainer.tsx:206]: this is a cool function, why nullish?

File: superset-frontend/src/dashboard/stylesheets/dashboard.less [hunk 3]
```diff
@@ -19,6 +19,7 @@
 /* header has mysterious extra margin */
 header.top {
   margin-bottom: 2px;
+  z-index: 10;
```
Comment (AAfghahi) [inline superset-frontend/src/dashboard/stylesheets/dashboard.less:22]: why is this necessary?

**Review by AAfghahi — APPROVED** [review]
Looks good! Just some questions, no feedback

File: superset-frontend/src/dashboard/stylesheets/dashboard.less (hunk 3 above)
Comment (kgabryje) [reply superset-frontend/src/dashboard/stylesheets/dashboard.less:22]: The main menu items on the right were not clickable because in edit mode the right panel was covering them (they were visible but unclickable because of invisible part of that panel covering them)

File: superset-frontend/src/addSlice/AddSliceContainer.tsx (hunk 2 above)
Comment (kgabryje) [reply superset-frontend/src/addSlice/AddSliceContainer.tsx:206]: `!dashboardId` would exclude `dashboardId === 0`, which is a valid id 😄
